src/todo-list-core/Data/TodoDbContext.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using TodoListAPI.Domain;
+using TodoListAPI.Interfaces;
+
+namespace TodoListAPI.Data
+{
+    public class TodoDbContext : DbContext
+    {
+        public DbSet<TodoList> TodoLists { get; set; }
+
+        public TodoDbContext(DbContextOptions options, ISessionInfo session)
+            : base(options)
+        {
+            _session = session ?? throw new ArgumentNullException(nameof(session));
+        }
+
+        #region Overrides of DbContext
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            Assembly executingAssembly = Assembly.GetExecutingAssembly();
+            modelBuilder.ApplyConfigurationsFromAssembly(executingAssembly);
+
```

Comment:
creating EF Core shadow properties by scanning the code and finding Domain entities. Using a mark interface to find them `IDomainEntity`